services/wallet/router/router.go

```diff
@@ -140,6 +143,48 @@ func (r *Router) SetTestBalanceMap(balanceMap map[string]*big.Int) {
 	}
 }
 
+func (r *Router) setCustomTxDetails(pathTxIdentity *requests.PathTxIdentity, pathTxCustomParams *requests.PathTxCustomParams) error {
+	r.activeRoutesMutex.Lock()
+	defer r.activeRoutesMutex.Unlock()
+	if r.activeRoutes == nil || len(r.activeRoutes.Best) == 0 {
+		return ErrCannotCustomizeIfNoRoute
+	}
+
+	for _, path := range r.activeRoutes.Best {
+		if path.PathIdentity() != pathTxIdentity.PathIdentity() {
+			continue
+		}
+
+		r.lastInputParamsMutex.Lock()
+		r.lastInputParams.PathTxCustomParams[pathTxIdentity.TxIdentityKey()] = pathTxCustomParams
+		r.lastInputParamsMutex.Unlock()
+
+		return nil
+	}
+
+	return ErrCannotFindPathForProvidedIdentity
+}
+
+func (r *Router) SetFeeMode(ctx context.Context, pathTxIdentity *requests.PathTxIdentity, feeMode fees.GasFeeMode) error {
+	if feeMode == fees.GasFeeCustom {
+		return ErrCustomFeeModeCannotBeSetThisWay
+	}
+
+	return r.setCustomTxDetails(pathTxIdentity, &requests.PathTxCustomParams{GasFeeMode: feeMode})
+}
+
+func (r *Router) SetCustomTxDetails(ctx context.Context, pathTxIdentity *requests.PathTxIdentity, pathTxCustomParams *requests.PathTxCustomParams) error {
```

Comment:
Could we find a better name for the private function? IMO a bit unclear

Reply:
I am open to suggestions, but it's not private, it's exposed via API.
There are 2 new functions that we're exposing, mentioned in the description of this PR, under "New endpoints added:" section.

Reply:
yes this one is not private but there is also setCustomTxDetails
so 2 functions with same name one private, on public
I didn't suggest cause nothing comes to mind but ... there is probably a better naming